src/components/TableV2/index.tsx

```diff
@@ -0,0 +1,325 @@
+import { IPaginatedResponse, IRowSection } from '@/types/index';
+import { setQueryAndRouter } from '@/utils';
+import { useDidUpdateEffect } from '@/utils/hooks';
+import { processRowSectionsLayout } from '@/utils/table';
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+import { BsFillArrowUpCircleFill } from 'react-icons/bs';
+import { IoReloadSharp } from 'react-icons/io5';
+import { useQuery } from 'react-query';
+// import { VscJson } from 'react-icons/vsc';
+import Pagination from '../Pagination';
+import { PaginationContainer } from '../Pagination/styles';
+import Skeleton from '../Skeleton';
+import Tooltip from '../Tooltip';
+import ExportButton from './ExportButton';
+import {
+  BackTopButton,
+  ContainerView,
+  EmptyRow,
+  ExportContainer,
+  FloatContainer,
+  HeaderItem,
+  IoReloadSharpWrapper,
+  ItemContainer,
+  LimitContainer,
+  LimitItems,
+  MobileCardItem,
+  MobileHeader,
+  RetryContainer,
+  TableBody,
+  TableContainer,
+  TableRow,
+  TableRowProps,
+} from './styles';
+
+export interface ITable {
+  type:
+    | 'transactions'
+    | 'blocks'
+    | 'accounts'
+    | 'assets'
+    | 'transactionDetail'
+    | 'buckets'
+    | 'accounts'
+    | 'assetsPage'
+    | 'holders'
+    | 'validators'
+    | 'validator'
+    | 'nodes'
+    | 'networkParams'
+    | 'proposals'
+    | 'votes'
+    | 'delegations'
+    | 'nfts'
+    | 'validatorsList'
+    | 'rewards'
+    | 'marketplaces'
+    | 'launchPad';
+
+  header: string[];
+  rowSections?: (item: any) => IRowSection[] | undefined;
+  dataName?: string;
+  request: (page: number, limit: number) => Promise<IPaginatedResponse>;
+  interval?: number;
+  intervalController?: React.Dispatch<React.SetStateAction<number>>;
+  showLimit?: boolean;
+  Filters?: React.FC;
+}
+
+const onErrorHandler = () => {
+  return {
+    onError: (err: unknown): void => {
+      console.error(err);
+    },
+    retry: 3,
+  };
+};
+
+const Table: React.FC<ITable> = ({
+  type,
+  header,
+  rowSections,
+  request,
+  dataName,
+  interval,
+  intervalController,
+  Filters,
+}) => {
+  const router = useRouter();
+  const { isMobile, isTablet } = useMobile();
+  const limits = [5, 10, 50, 100];
+  const [scrollTop, setScrollTop] = useState<boolean>(false);
+
+  const tableRef = React.useRef<HTMLDivElement>(null);
+
+  const page = Number(router.query?.page) || 1;
+  const limit = Number(router.query?.limit) || 10;
+
+  const tableRequest = async (page: number, limit: number): Promise<any> => {
+    let responseFormatted = {};
+    try {
+      const response = await request(page, limit);
+      if (!response.error && dataName) {
+        responseFormatted = {
+          items: response.data[dataName],
+          totalPages: response?.pagination?.totalPages,
+        };
+        return responseFormatted;
+      }
+      // setPage(1);
+      return [];
+    } catch (error) {
+      console.error(error);
+    }
```

Comment:
In others places we call toast also, here shouldn't?

Reply:
if it catches an error, it will render an empty table. In an empty table, the user can refetch